### PR TITLE
feat: virtual list with absolute positions for new items

### PR DESCRIPTION
**Pull request by alexis-kochetov**

### CODE REVIEW

File: .run/App.Server.run.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <option name="PROJECT_WORKING_DIRECTORY_TRACKING" value="1" />
     <option name="PROJECT_KIND" value="DotNetCore" />
     <option name="PROJECT_TFM" value="net6.0" />
-    <browser url="https://localhost:7081/chat/the-actual-one" name="98ca6316-2f89-46d9-a9e5-fa9e2b0625b3" start="true" />
+    <browser url="https://local.actual.chat/chat/the-actual-one" name="98ca6316-2f89-46d9-a9e5-fa9e2b0625b3" />
```
Comment (FrolYo) [inline .run/App.Server.run.xml:19]: why are we trying to replace localhost with local.actual.chat? does local.actual.chat support page autoreload on connection restore?

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/ChatView.razor.cs [hunk 2]
```diff
@@ -127,14 +123,17 @@ private async Task MonitorVisibleKeyChanges(CancellationToken cancellationToken)
...
+    async Task<VirtualListData<ChatMessageModel>> IVirtualListDataSource<ChatMessageModel>.GetData(
         VirtualListDataQuery query,
+        VirtualListData<ChatMessageModel> oldData,
         CancellationToken cancellationToken)
     {
         await WhenInitialized;
 
         var chat = Chat;
         var chatId = chat.Id.Value;
+        var currentAuthor = await ChatAuthors.Get(Session, chatId, cancellationToken);
```
Comment (FrolYo) [inline src/dotnet/Chat.UI.Blazor/Components/ChatView/ChatView.razor.cs:135]: I was told that we do not almost use `current` as a useless prefix

**Review by FrolYo — APPROVED** [review]

File: .run/App.Server.run.xml (hunk 1 above)
Comment (alexis-kochetov) [reply .run/App.Server.run.xml:19]: `local.actual.chat` uses nginx so image proxy works locally as well. Additionally notifications also works when the site is not localhost

Comment (alexis-kochetov) [reply .run/App.Server.run.xml:19]: Why do you expect it doesn't reload on conneciton restore?

File: src/dotnet/Chat.UI.Blazor/Components/ChatView/ChatView.razor.cs (hunk 2 above)
Comment (alexis-kochetov) [reply src/dotnet/Chat.UI.Blazor/Components/ChatView/ChatView.razor.cs:135]: OK

File: .run/App.Server.run.xml (hunk 1 above)
Comment (FrolYo) [reply .run/App.Server.run.xml:19]: Nginx will show you 502 error if local dev host is stopped. 502 is not automatically reloaded when host is restarted.

Anyways, I think it's such a minor inconvenience comparing to benefits you mentioned above

**Pull request merged**